### PR TITLE
docs: Less confusing documentation for `torpassword`

### PR DESCRIPTION
**Pull request by blob42**
The current documentation leads the reader to think `hash-password` is an other option. 
This change is less confusing and make it clear how to use this option.

### CODE REVIEW

**Comment by ch4ot1c** [comment]
As a note, `--hash-password` comes from here: https://www.torproject.org/docs/tor-manual.html.en#opt-hash-password

Concept ACK

File: doc/tor.md [hunk 1]
```diff
@@ -110,8 +110,8 @@ are seen with `-debug=tor` they can be resolved by adding both the user running
 the user running bitcoind to the same group and setting permissions appropriately. On
 Debian-based systems the user running bitcoind can be added to the debian-tor group,
 which has the appropriate permissions. An alternative authentication method is the use
-of the `-torpassword` flag and a `hash-password` which can be enabled and specified in
-Tor configuration.
+of the `-torpassword=hashpassword` option. The `hashpassword` can be enabled and specified in
+Tor configuration under the `HashedControlPassword` option.
```
Comment (laanwj) [inline doc/tor.md]: it's an improvement but still not fully clear as to what actually needs to be done to set up tor password authentication. to be clear:
- bitcoind's `torpassword` takes the *unhashed* password
- tor's configuration `HashedControlPassword` has a hashed password

the way that this is worded, it sounds like bitcoind's argument takes the hashes password too

Comment (sp4ke) [reply doc/tor.md]: I just pushed an other change that adds more details on the relation between `password` and `HashedControlPassword`.

**Comment by fanquake** [comment]
@sp4ke could you please [squash your commits](https://github.com/bitcoin/bitcoin/blob/master/CONTRIBUTING.md#squashing-commits), then this can be merged.

**Comment by fanquake** [comment]
Rebased and squashed in #14804.

**Comment by sp4ke** [comment]
@fanquake sorry I missed the notification

**Pull request closed without merging**